### PR TITLE
DOC: Document full capabilities of aggfunc argument

### LINKED ISSUE #2302: DOC: `dissolve`: Document full capabilities of `aggfunc` argument
The documentation of `GeoDataFrame.dissolve` should clarify that `aggfunc` argument supports all the capabilities of 
Pandas [`DataFrame.agg`](geopandas.GeoDataFrame.dissolve).

These options are often quite useful, as one may want to produce multiple statistics per column, or different statistics for each column.

Currently, the [User Guide](https://geopandas.org/en/stable/docs/user_guide/aggregation_with_dissolve.html) only contains examples of passing a string to `aggfunc`. (It mentions that "it also accepts other summary statistic options as allowed by pandas.groupby", however the provided link to the `groupby` documentation isn't particularly useful.
It should also show the other options, and give an example.

The [API Reference](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoDataFrame.dissolve.html#geopandas.GeoDataFrame.dissolve) states that `aggfunc` must be a function or string. It misses the option to pass a list or dictionary.
It should also show the other options

### PR DESCRIPTION
**Pull request by oefe**
Resolves #2302

### CODE REVIEW

File: doc/source/docs/user_guide/aggregation_with_dissolve.rst [hunk 1]
```diff
@@ -75,3 +75,23 @@ However it also accepts other summary statistic options as allowed by :meth:`pan
...
+and the ``'pop_est'`` column using ``'min'`` and ``'max'``:
+
+.. ipython:: python
+
+   world = geopandas.read_file(geopandas.datasets.get_path('naturalearth_lowres'))
+   continents = world.dissolve(by='continent', aggfunc=
+   {
+      'name': 'count',
+      'pop_est': ['min', 'max'],
+   })
```
Comment (martinfleis) [inline doc/source/docs/user_guide/aggregation_with_dissolve.rst:95]:
```suggestion
    world = geopandas.read_file(geopandas.datasets.get_path("naturalearth_lowres"))
    continents = world.dissolve(
        by="continent",
        aggfunc={
            "name": "count",
            "pop_est": ["min", "max"],
        },
    )
```

I've just applied `black` formatting here for consistency and better readability.

**Review by martinfleis — COMMENTED** [review]
Thanks! This is certainly super useful. I just left one minor formatting suggestion.

File: doc/source/docs/user_guide/aggregation_with_dissolve.rst (hunk 1 above)
Comment (oefe) [reply doc/source/docs/user_guide/aggregation_with_dissolve.rst]: Thanks!
(BTW, is there a way to invoke black on code within RST files?)

**Review by martinfleis — APPROVED** [review]
Thanks!

**Pull request merged**